### DISCUSSION TITLE
udp send on windows happens immediately, not in loop

### DISCUSSION THREAD

**Discussion by BMBurstein** [libuv/libuv#3359] [Q&A] [answered]
* **Version**: 1.39
* **Platform**: Windows 10

I call uv_udp_send before executing uv_run, and I expected the actual sending to not occur until the loop was running, but it was sent immediately. Or am I using it wrong?

**Comment by bnoordhuis** [comment, marked as answer]
No, that's right. If the datagram can be sent right away without blocking (and it usually can), then that's what libuv will do. It results in lower latency and (if you use `uv_udp_try_send()`) lower memory consumption.

**Reply by BMBurstein** [reply]
Is this documented? Because it threw me off and wasted some time trying to debug the wrong thing. It caused my server to send the response before my listener was ready

**Comment by bnoordhuis** [comment]
In the specific case of `uv_udp_try_send()`, yes, but in general: no.

Libuv doesn't make promises when it performs operations, that would be needlessly constraining, only that it won't block.

**Reply by BMBurstein** [reply]
Ok. I thought there was an implicit promise to only run requests once the loop was started.
